File: 1월 1주차/징검다리 건너기.py
def go(arr, m, k):
    cnt = 1
    for i in arr:
        if i >= m:
            cnt = 1
        else:
            cnt += 1
            if cnt > k:
                return True
    return False


def solution(stones, k):
    l,r = 0, max(stones)
    while l <= r:
        m = (l+r)//2
        if go(stones, m, k):
            r = m - 1
        else:
            l = m + 1
    return r
            
        

File: 5월 1주차/징검다리 건너기.js
function check(stones,m,k){
    let cnt = 0;
    for (let i=0; i<stones.length; i++){
        if (stones[i] >= m) cnt = 0
        else cnt++
        if (cnt>=k) return false
    }
    return true
}

function solution(stones, k) {
    var answer = 0;
    let l=0, r=200000000;
    while (l<=r){
        let m = Math.floor((l+r)/2);
        if(check(stones,m,k)){
            l = m+1
        } else{
            r = m-1
        }
    }
    
    return r;
}